src/components/message.js

```diff
@@ -9,11 +9,70 @@ var IntlMessage = React.createClass({
     displayName: 'IntlMessage',
     mixins     : [IntlMixin],
 
+    propTypes: {
+        tagName: React.PropTypes.string,
+        message: React.PropTypes.string.isRequired
+    },
+
+    getDefaultProps: function () {
+        return {tagName: 'span'};
+    },
+
     render: function () {
         var props   = this.props;
-        var message = props.children;
+        var tagName = props.tagName;
+        var message = props.message;
+
+        // Creates a token with a random guid that should not be guessible or
+        // conflict with other parts of the `message` string.
+        var guid     = Math.floor(Math.random() * 0x10000000000).toString(16);
+        var token    = '@__ELEMENT_' + guid + '__@';
+        var elements = [];
+
+        // Iterates over the `props` to keep track of any React Element values
+        // so they can be represented by the `token` as a placeholder when the
+        // `message` is formatted. This allows the formatted message to then be
+        // broken-up into parts with references to the React Elements inserted
+        // back in.
+        var values = Object.keys(props).reduce(function (values, name) {
+            var value = props[name];
+
+            if (React.isValidElement(value)) {
+                values[name] = token;
+                elements.push(value);
```

Comment:
Realized there's a bug here since there's no reference to _which_ elements should be inserted back _where_.